src/lib/optimizer/strategy/predicate_pushdown_rule.cpp

```diff
@@ -0,0 +1,111 @@
+#include "predicate_pushdown_rule.hpp"
+#include "all_parameter_variant.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/lqp_column_reference.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+
+namespace opossum {
+
+std::string PredicatePushdownRule::name() const { return "Predicate Pushdown Rule"; }
+
+bool PredicatePushdownRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  // find predicate node
+  if (node->type() == LQPNodeType::Predicate) {
+    const auto parents = node->parents();
+    if (parents.empty() || parents.size() > 1) {
+      return false;
+    }
+
+    const auto predicate_node = std::dynamic_pointer_cast<PredicateNode>(node);
+
+    // can predicate node demoted?
```

Comment:
I don't like "demote" too much in this context. It is the push down rule - why don't we say "pushed down" here?